src/IECoreDelight/Renderer.cpp

```diff
@@ -986,8 +986,42 @@ IECore::InternedString g_frameOptionName( "frame" );
 IECore::InternedString g_cameraOptionName( "camera" );
 IECore::InternedString g_sampleMotionOptionName( "sampleMotion" );
 IECore::InternedString g_oversamplingOptionName( "dl:oversampling" );
+
+IECore::InternedString g_maxLengthDiffuseOptionName( "dl:maximumraylength.diffuse" );
+IECore::InternedString g_maxLengthHairOptionName( "dl:maximumraylength.hair" );
+IECore::InternedString g_maxLengthReflectionOptionName( "dl:maximumraylength.reflection" );
+IECore::InternedString g_maxLengthRefractionOptionName( "dl:maximumraylength.refraction" );
+IECore::InternedString g_maxLengthSpecularOptionName( "dl:maximumraylength.specular" );
+IECore::InternedString g_maxLengthVolumeOptionName( "dl:maximumraylength.volume" );
+IECore::InternedString g_clampIndirectOptionName( "dl:clampindirect" );
+IECore::InternedString g_showMultipleScatteringOptionName( "dl:show.multiplescattering" );
+
 const char *g_screenHandle = "ieCoreDelight:defaultScreen";
 
+ConstDataPtr conformOptionValue( const InternedString &name, const Object *value )
+{
+	if(
+		name == g_maxLengthDiffuseOptionName ||
+		name == g_maxLengthHairOptionName ||
+		name == g_maxLengthReflectionOptionName ||
+		name == g_maxLengthRefractionOptionName ||
+		name == g_maxLengthSpecularOptionName ||
+		name == g_maxLengthVolumeOptionName ||
+		name == g_clampIndirectOptionName
+	)
+	{
+		// These parameters fail to set properly if they are not doubles
+		auto data = runTimeCast<const FloatData>( value );
```

Comment:
This will crash if someone does provide DoubleData, because we're not checking for a null return from the cast. I think we might as well just return the original data if the cast fails (if it's double then great, if not, we'll give the wrong type to NSI and the caller will find out).

Reply:
> we're not checking for a null return from the cast

Fixed in https://github.com/GafferHQ/gaffer/pull/5333/commits/166474356a9cc4f9db52b528a734678df939b569

> might as well just return the original data if the cast fails

I'm not doing that in this commit, because I believe the same behavior can be accomplished by returning `nullptr` because of the check before setting the option of `value = conformedValue ? conformedValue.get() : value;` This way we avoid creating a new shared pointer to wrap the original value (though I suppose probably not much a performance problem just for setting options).

I think my logic is sound on that?